components/at_mention/at_mention.tsx

```diff
@@ -84,40 +84,43 @@ export default class AtMention extends React.PureComponent<Props, State> {
         }
 
         const suffix = this.props.mentionName.substring(user.username.length);
+        const displayName = displayUsername(user, this.props.teammateNameDisplay);
 
-        let className = 'mention-link';
-        if (!this.props.disableHighlight && user.id === this.props.currentUserId) {
-            className += ' mention--highlight';
-        }
+        const highlightMention = !this.props.disableHighlight && user.id === this.props.currentUserId;
 
         return (
-            <span>
-                <Overlay
-                    placement={this.state.placement}
-                    show={this.state.show}
-                    target={this.state.target}
-                    rootClose={true}
-                    onHide={this.hideOverlay}
-                >
-                    <ProfilePopover
-                        className='user-profile-popover'
-                        userId={user.id}
-                        src={Client4.getProfilePictureUrl(user.id, user.last_picture_update)}
-                        isRHS={this.props.isRHS}
-                        hasMention={this.props.hasMention}
-                        hide={this.hideOverlay}
-                        channelId={this.props.channelId}
-                    />
-                </Overlay>
-                <a
-                    className={className}
-                    onClick={this.handleClick}
-                    ref={this.overlayRef}
+            <>
+                <span
+                    className={highlightMention ? 'mention--highlight' : ''}
```

Comment:
nit: we can add null here or pass the condition in another way, as this updates the snapshots and renders an empty string classname.